airflow/api_fastapi/execution_api/datamodels/taskinstance.py

```diff
@@ -276,6 +276,9 @@ class TIRunContext(BaseModel):
     xcom_keys_to_clear: Annotated[list[str], Field(default_factory=list)]
     """List of Xcom keys that need to be cleared and purged on by the worker."""
 
+    should_retry_on_failure: bool
```

Comment:
This is a bit verbose -- maybe `eligble_for_retry` or `final_attempt` (this one would invert the meaning clearly)

Reply:
`do_retry` or `eligible_for_retry` i'd say